### PR TITLE
chore(docs): update new datasource developer docs

### PR DESCRIPTION
**Pull request by blackboxsw**
Adding a bit more context to new datasource developer docs having  mentioned a few of these details in review suggestions on new datasource additions for WSL, CloudCIX and Aeza.net.

## Proposed Commit Message
```
    chore(docs): update new datasource developer docs
    
    Use links instead of :file: for repo-local files to ensure CI
    validates linkcheck on those files in the event that upstream changes
    their names or location in the future.
    
    Reorder ds-identify docs above python DataSource module as the
    python module needs to reflect the same ds-identify detection logic
    in a ds_detect module.
    
    Add that ds-identify does rely on identification alternatives to DMI such as
    attached device labels, kernel command line and virt type.
    Add example reference to simple NWCS datasource commit for ds-identify docs.
```

## Test Steps

tox -e doc
xdg-open doc/rtd_html/development/datasource_creation.html

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: doc/rtd/development/datasource_creation.rst [hunk 1]
```diff
@@ -118,33 +138,19 @@ Add tests for datasource module
 -------------------------------
 
 Add a new file with some tests for the module to
-:file:`cloudinit/sources/test_<yourplatform>.py`. For example, see
-:file:`cloudinit/sources/tests/test_oracle.py`
-
-Update ``ds-identify``
-----------------------
-
-In ``systemd`` systems, ``ds-identify`` is used to detect which datasource
```
Comment (blackboxsw) [inline doc/rtd/development/datasource_creation.rst:127]: Added mention of Alpine and BSD too as this is broader than just systemd nowadays

File: doc/rtd/development/datasource_creation.rst [hunk 2]
```diff
@@ -86,25 +88,43 @@ There are multiple ways to provide `user data`, `metadata`, and
...
+
 Add datasource module cloudinit/sources/DataSource<CloudPlatform>.py
 --------------------------------------------------------------------
 
-We suggest you start by copying one of the simpler datasources
-such as ``DataSourceHetzner``.
+Use one of the simpler datasources such as ``DataSourceHetzner`` as a guiding
+template for style and expectations. The DataSource module should implement a
+``ds_detect`` method validates the same identification conditions defined
+in ds-identify and returns ``True`` when met. This allows cloud-init to support
```
Comment (blackboxsw) [inline doc/rtd/development/datasource_creation.rst:118]: @blackboxsw add netv2 discussion here. from Alberto

**Review by TheRealFalcon — APPROVED** [review]
One inline nit, but LGTM

File: doc/rtd/development/datasource_creation.rst [hunk 3]
```diff
@@ -29,17 +29,19 @@ cloud-init support.
...
+A mechanism for identification
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Each cloud platform must positively identify itself to the guest. This allows
 the guest to make educated decisions based on the platform on which it is
 running. On the x86 and arm64 architectures, many clouds identify themselves
 through `DMI`_ data. For example, Oracle's public cloud provides the string
-``'OracleCloud.com'`` in the DMI chassis-asset field.
+``'OracleCloud.com'`` in the DMI chassis-asset field. Some platforms present
+attached devices with well known filesystem label, kernel command line flags or
```
Comment (aciba90) [inline doc/rtd/development/datasource_creation.rst:43]: Either

```suggestion
attached devices with a well known filesystem label, kernel command line flags or
```

or 

```suggestion
attached devices with well known filesystem labels, kernel command line flags or
```

right?

File: doc/rtd/development/datasource_creation.rst [hunk 4]
```diff
@@ -86,25 +88,43 @@ There are multiple ways to provide `user data`, `metadata`, and
...
 will need to do all of the following:
 
+Update ``ds-identify``
+----------------------
+
+In ``systemd``, Alpine and BSD environments, ``ds-identify`` runs in early boot
+to detect which datasource should be enabled, or if ``cloud-init`` should run
+at all. You'll need to add early identification support for the platform via a
+``dscheck_<CloudPlatform>`` function in `tools/ds-identify`_.
+For example, see `NWCS support`_
```
Comment (aciba90) [inline doc/rtd/development/datasource_creation.rst:104]:
```suggestion
For example, see `NWCS support`_.
```

File: doc/rtd/development/datasource_creation.rst [hunk 5]
```diff
@@ -86,25 +88,43 @@ There are multiple ways to provide `user data`, `metadata`, and
...
+
+Add relevant tests in a new class to
+`tests/unittests/test_ds_identify.py`_. Use ``TestOracle`` as an example.
+
 Add datasource module cloudinit/sources/DataSource<CloudPlatform>.py
 --------------------------------------------------------------------
 
-We suggest you start by copying one of the simpler datasources
-such as ``DataSourceHetzner``.
+Use one of the simpler datasources such as ``DataSourceHetzner`` as a guiding
```
Comment (aciba90) [inline doc/rtd/development/datasource_creation.rst:115]:
```suggestion
Use one of the simpler datasources, such as ``DataSourceHetzner``, as a guiding
```

File: doc/rtd/development/datasource_creation.rst (hunk 2 above)
Comment (aciba90) [reply doc/rtd/development/datasource_creation.rst:118]: I do not see changes covering this. Maybe they got mangled by some git operation?

**Review by aciba90 — CHANGES_REQUESTED** [review]
Thanks for improving these docs!

**Pull request closed without merging**